src/crimson/os/seastore/seastore.h

```diff
@@ -78,47 +92,482 @@ class SeaStore final : public FuturizedStore {
+      internal_context_t &ctx,
+      OnodeRef &onode,
+      std::string &&name);
+    tm_ret _xattr_clear(
+      internal_context_t &ctx,
+      OnodeRef &onode);
+    tm_ret _create_collection(
+      internal_context_t &ctx,
+      const coll_t& cid, int bits);
+    tm_ret _remove_collection(
+      internal_context_t &ctx,
+      const coll_t& cid);
+    using omap_set_kvs_ret = tm_iertr::future<>;
+    omap_set_kvs_ret _omap_set_kvs(
+      OnodeRef &onode,
+      const omap_root_le_t& omap_root,
+      Transaction& t,
+      omap_root_le_t& mutable_omap_root,
+      std::map<std::string, ceph::bufferlist>&& kvs);
+
+    boost::intrusive_ptr<SeastoreCollection> _get_collection(const coll_t& cid);
+
+    static constexpr auto LAT_MAX = static_cast<std::size_t>(op_type_t::MAX);
+    struct {
+      std::array<seastar::metrics::histogram, LAT_MAX> op_lat;
+    } stats;
+
+    seastar::metrics::histogram& get_latency(op_type_t op_type) {
+      assert(static_cast<std::size_t>(op_type) < stats.op_lat.size());
+      return stats.op_lat[static_cast<std::size_t>(op_type)];
+    }
+
+    void add_latency_sample(op_type_t op_type,
+         std::chrono::steady_clock::duration dur) {
+      seastar::metrics::histogram& lat = get_latency(op_type);
+      lat.sample_count++;
+      lat.sample_sum += std::chrono::duration_cast<std::chrono::milliseconds>(dur).count();
+    }
+    seastar::metrics::metric_group metrics;
+    void register_metrics();
+  };
+
+  seastar::sharded<ShardStores> shard_stores;
+
+  seastar::future<> shard_mkfs(
+    std::vector<secondary_device_set_t> &sds,
+    uuid_d new_osd_fsid);
+  seastar::future<> sec_shard_mkfs(
+    const std::string path,
+    device_type_t dtype,
+    device_id_t id,
+    std::vector<secondary_device_set_t> &sds,
+    uuid_d new_osd_fsid);
+
+public:
   SeaStore(
     const std::string& root,
     MDStoreRef mdstore,
     DeviceRef device,
     bool is_test);
   ~SeaStore();
-    
+
+  seastar::future<> start() final;
   seastar::future<> stop() final;
-  mount_ertr::future<> mount() final;
-  seastar::future<> umount() final;
+
+  mount_ertr::future<> mount() final {
+    return shard_stores.invoke_on_all(
+      [](auto &local_store) {
+      return local_store.mount().handle_error(
+        crimson::ct_error::assert_all{
+        "Invalid error in SeaStore::mount"
+      });
+    });
+  }
+  seastar::future<> umount() final {
+    return shard_stores.invoke_on_all(
+      [](auto &local_store) {
+      return local_store.umount();
+    });
+  }
 
   mkfs_ertr::future<> mkfs(uuid_d new_osd_fsid) final;
-  seastar::future<store_statfs_t> stat() const final;
+  seastar::future<store_statfs_t> stat() final;
+  seastar::future<std::vector<coll_core_t>> list_collections() final;
+  seastar::future<std::tuple<int, std::string>> read_meta(const std::string& key) final;
+
+  unsigned get_max_attr_name_length() const final {
+    return 256;
+  }
+  seastar::future<> write_fsid(uuid_d new_osd_fsid);
```

Comment:
Move it to the private section.

Please also scan the code to move other methods to the private section if it is not supposed to be called by OSD.